wfe/wfe.go

```diff
@@ -1882,6 +1891,18 @@ func (wfe *WebFrontEndImpl) FinalizeOrder(
 		return
 	}
 
+	// Ensure the signature on the CSR is good
+	if err := parsedCSR.CheckSignature(); err != nil {
```

Comment:
We prefer to avoid the `if err := ...; err != nil` syntax: https://github.com/letsencrypt/boulder/blob/main/docs/CONTRIBUTING.md#error-handling

```suggestion
	err = parsedCSR.CheckSignature()
	if err != nil {
```